.github/workflows/nodejs.yml

```diff
@@ -0,0 +1,60 @@
+name: Node.js integration
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+  workflow_dispatch:
+
+jobs:
+  integration:
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [macos-13, macos-latest, ubuntu-latest, windows-latest]
+        python: ["3.8", "3.10", "3.12"]
+
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Clone gyp-next
+        uses: actions/checkout@v4
+        with:
+          path: gyp-next
+      - name: Clone nodejs/node
+        uses: actions/checkout@v4
+        with:
+          repository: nodejs/node
+          path: node
+      - uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python }}
+          allow-prereleases: true
+
+      # macOS and Linux
+      - name: Replace gyp in Node.js
+        if: runner.os != 'Windows'
```

Comment:
Bash is the default on Linux and macOS so the `shell: bash` line is only useful on Windows.  This command will work on all three platforms with the `shell: bash` line.
```suggestion
```